include/fmt/core.h

```diff
@@ -363,6 +363,9 @@ template <typename Char> class basic_string_view {
     the size with ``std::char_traits<Char>::length``.
     \endrst
    */
+#ifdef FMT_USE_STRING_VIEW  // C++17's char_traits::length() is constexpr.
```

Comment:
`FMT_USE_STRING_VIEW` is not a good indicator of whether `char_traits::length()` is `constexpr` because libc++ supports `string_view` in pre-c++17.

Reply:
While that may be true, it's not guaranteed by the standard to be constexpr prior to the C++17 language standard as far as I can tell. Prior to my change this function was _never_ marked constexpr, so I think this change is still an improvement. If you have a more direct suggestion I'd be happy to wrap it in, as I'm not sure how to evaluate based on what C++ standard library version is being used and what particular version they each decided to support this.

Reply:
Since it's a C++17 feature, a better check would be `__cplusplus >= 201703L`.